db/migrate/20240423054323_create_flags.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class CreateFlags < ActiveRecord::Migration[7.0]
+  def change
+    create_table :flags do |t|
+      t.string :name, unique: true
```

Comment:
I feel like the `name` here should be the `name_key` in the database, and we should have a separate `name` column which would be the real, human-readable name for the custom flag. We would calculate the `name_key` when the flag is created and ensure it does not overlap with any of the system flags (which would have an empty `name` column since we have translations for them).

We also need a `description` column here for the custom flags.